### PR TITLE
FEATURE: Add newly loaded topics to bulk-select automatically if Select All is on

### PR DESCRIPTION
**Pull request by OsamaSayegh**
Currently it's very tedious to bulk select hundreds of topics in a topic list -- each time a new batch of topics is loaded you have to scroll all the way to the top to click the `Select All` button and scroll back down to load the next batch, or you have to tick each topic individually.

This PR should make that process a lot easier because we will now remember if the `Select All` button was clicked and so whenever a new batch of topics is loaded, it'll be automatically added to bulk select (if bulk select is activated of course).

Meta topic: https://meta.discourse.org/t/add-select-all-controls-at-the-bottom-of-the-list/178020/2?u=osama.

### CODE REVIEW

**Review by eviltrout — COMMENTED** [review]
That's a lot of code without any tests. I understand parts of this PR are hard to test because they depend on things like scrolling but maybe you could add a couple small tests for the pieces that aren't so hard?

**Comment by OsamaSayegh** [comment]
Ok, I've added a few tests. Let me know if there is anything else 😊

File: app/assets/javascripts/discourse/tests/integration/components/topic-list-test.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+} from "discourse/tests/helpers/component-test";
+import { discourseModule } from "discourse/tests/helpers/qunit-helpers";
+import Topic from "discourse/models/topic";
+import hbs from "htmlbars-inline-precompile";
+
+discourseModule("Integration | Component | topic-list", function (hooks) {
+  setupRenderingTest(hooks);
+
+  componentTest("bulk select", {
+    template: hbs`{{topic-list
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/tests/integration/components/topic-list-test.js:13]: minor but indenting looks off here

File: app/assets/javascripts/discourse/app/models/topic-list.js [hunk 2]
```diff
@@ -92,10 +92,11 @@ const TopicList = RestModel.extend({
 
       return ajax({ url: moreUrl }).then((result) => {
         let topicsAdded = 0;
+        let newTopics = [];
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/app/models/topic-list.js:95]: I'm not sure if we need this change?

**Comment by tgxworld** [comment]
The change looks OK to me but I'm alittle rusty with Javascript and Ember. Maybe @eviltrout can have another look?

File: app/assets/javascripts/discourse/app/models/topic-list.js (hunk 2 above)
Comment (OsamaSayegh) [reply app/assets/javascripts/discourse/app/models/topic-list.js:95]: It needs to be in this scope to allow the `return` line below to see it (unlike Ruby, `if` does have its own scope in JS)

Comment (tgxworld) [reply app/assets/javascripts/discourse/app/models/topic-list.js:95]: The `return` below is within the `if` scope as well so we shouldn't have to declare the variable outside of the scope.

Comment (OsamaSayegh) [reply app/assets/javascripts/discourse/app/models/topic-list.js:95]: Looks like I need glasses lol, will fix it thanks!

**Review by eviltrout — APPROVED** [review]
This looks good to me, thanks!

**Pull request merged**